WIP: deploying esco-content-menu from quickstart

jgribonvald opened the pull request:
<!--
Thank you for your pull request. Please review below requirements.

Bug fixes and new features should be reported on the issue tracker: https://issues.jasig.org/browse/UP/

Contributors guide: https://github.com/Jasig/uPortal/blob/master/CONTRIBUTING.md
-->

##### Checklist
<!-- Remove items that do not apply. For completed items, change [ ] to [x]. -->

-   [x] the [individual contributor license agreement][] is signed
-   [x] commit message follows [commit guidelines][]
-   [ ] tests are included
-   [ ] documentation is changed or added
-   [ ] [message properties][] have been updated with new phrases
-   [ ] view conforms with [WCAG 2.0 AA][]

##### Description of change
<!-- Provide a description of the change below this comment. -->
Here is a commit on how to deploy the esco-content-menu.

How would you like to include it ? Adding only a documentation into esco-content-menu wc ? or like that ?
After I'm agree the script into the jsp invoker can be reviewed, but I did like that as I wasn't able to deploy it with the menu button component, I need to generate the button directly into the body and adding an event javascript to open/hidde the page.

So this commit is in WIP morefor your feedback !

<!-- Reference Links -->

[individual contributor license agreement]: https://github.com/Jasig/uPortal/blob/master/.github/CONTRIBUTING.md#individual-contributor-license-agreement
[commit guidelines]: https://github.com/Jasig/uPortal/blob/master/.github/CONTRIBUTING.md#commit
[message properties]: https://github.com/Jasig/uPortal/tree/master/uportal-war/src/main/resources/properties/i18n
[WCAG 2.0 AA]: https://www.w3.org/WAI/WCAG20/quickref/?levels=aaa&technologies=smil%2Cpdf%2Cflash%2Csl

ChristianMurphy commented:
related to #220

jgribonvald commented:
I think that I can't improve more things inside this PR, or tell me what do do ! (I shown how it was easy to deploy it !)
So you have the rigth to close this PR, or ??? 
it's one way of moving from the good old one Menu, but a part is missing: the accessibility check ;)

drewwills commented:
@jgribonvald -- I'm very interested in getting the ESCO Hamburger into quickstart.

This week is pretty busy, so it's hard to look, but I do think this is a good idea.

jgribonvald commented:
As example a quick and fast way to change the welcome page:
- you can use this same commit
- modify the jsp in this commit with a such code
```jsp
<esco-content-menu
      default-org-logo="${alternativeBanner[0]}"
      favorites-portlet-card-size="${favoritesPortletCardSize[0]}"
      grid-portlet-card-size="${gridPortletCardSize[0]}"
      hide-action-mode="${hideActionMode[0]}"
      portlet-api-url="${portletApiUrl[0]}"
      layout-api-url="${layoutApiUrl[0]}"
      organization-api-url="${organizationApiUrl[0]}"
      user-info-api-url="${userInfoApiUrl[0]}"
      :show-favorites-in-slider="false"
      debug>
      <div slot="header-buttons"></div>
      <div slot="content-user"></div>
    </esco-content-menu>
</div>

<script type="text/javascript">
  document.addEventListener("DOMContentLoaded", function(event) {
    const parent = document.getElementById("portalPageBody");
    while (parent.firstChild) {
      parent.firstChild.remove();
    }
    parent.appendChild(document.querySelector('#content-menu'));
  });
</script>
```
and you will be able to get a such portal content:
![image](https://user-images.githubusercontent.com/1597283/56287085-c2b27f80-611b-11e9-8c15-18cad4edc1db.png)

jgribonvald commented:
I've modified:
* the jsp to make optional to hide the native uPortal Nav from preference portlet property `hidePortalNav`
* changed the admin-dashboard to evoid to load several times the esco-content-menu lib and vuejs lib too as it's integrated from the esco-content-menu jsp on each page

I think if we want to integrate it, it can be done now. 
Maybe some change should be done on the hamburger menu for Accessibility.  Any help would be welcome on this part.
